Synchronize FileAPI WPT

annevk opened the pull request:
#### 0b25769a33bbdf56e9f725697a4e577ac8033478
<pre>
Synchronize FileAPI WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=274457">https://bugs.webkit.org/show_bug.cgi?id=274457</a>

Reviewed by Tim Nguyen.

Quite a few caveats unfortunately:

- Ignore a test change with Float16Array for now as it would reduce coverage.
- BlobURL/cross-partition.tentative.https.html is not imported for now.
- blob/Blob-methods-from-detached-frame.html is removed as we exported it as
  Blob-methods-from-detached-frame.html and imported that one at a later point.
  So in effect this deduplicates it.
- Many -expected.txt files are removed because --clean-dest-dir was not used
  during prior imports.
- A change to FileAPI/url/multi-global-origin-serialization.sub.html is ignored
  for now.

Latest upstream commit for this directory:
<a href="https://github.com/web-platform-tests/wpt/commit/5aa50dd4151b5bc1d04d5505366c6e27df30af5b">https://github.com/web-platform-tests/wpt/commit/5aa50dd4151b5bc1d04d5505366c6e27df30af5b</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-methods-from-detached-frame-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-methods-from-detached-frame.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-slice-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-slice-overflow-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream-byob-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream-sync-xhr-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any.js:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/File-constructor-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/send-file-formdata-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/send-file-formdata-utf-8-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/fileReader-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/Determining-Encoding-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/FileReader-event-handler-attributes-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/FileReader-multiple-reads-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_abort-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_error-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsArrayBuffer-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsBinaryString-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsDataURL-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsText-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readystate-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_result-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/sandboxed-iframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url-format.any.js:
(test.t.t.add_cleanup):
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url-with-fetch.any.js:
(async garbageCollect.else.gcRec): Deleted.
(async garbageCollect): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url_xmlhttprequest_img-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/279121@main">https://commits.webkit.org/279121@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e0ad5765e3676675f743b37906ff323e1e93945

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52546 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31880 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4972 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55820 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3269 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38422 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2970 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42696 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2083 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54643 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29540 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45348 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23787 "Found 1 new API test failure: /WPE/TestMultiprocess:/webkit/WebKitWebView/multiprocess-create-ready-close (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26709 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2626 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1428 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48593 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2782 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57416 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27680 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2808 "Found 4 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/animating-new-content-subset.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-offscreen-new.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-offscreen-old.html, workers/wasm-hashset-many.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50092 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28907 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45463 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49355 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11471 "Found 1 new test failure: workers/wasm-hashset-many.html (failure)") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29819 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7705 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28656 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please include upstream commit

annevk committed:
Synchronize FileAPI WPT
https://bugs.webkit.org/show_bug.cgi?id=274457

Reviewed by Tim Nguyen.

Quite a few caveats unfortunately:

- Ignore a test change with Float16Array for now as it would reduce coverage.
- BlobURL/cross-partition.tentative.https.html is not imported for now.
- blob/Blob-methods-from-detached-frame.html is removed as we exported it as
  Blob-methods-from-detached-frame.html and imported that one at a later point.
  So in effect this deduplicates it.
- Many -expected.txt files are removed because --clean-dest-dir was not used
  during prior imports.
- A change to FileAPI/url/multi-global-origin-serialization.sub.html is ignored
  for now.

Latest upstream commit for this directory:
https://github.com/web-platform-tests/wpt/commit/5aa50dd4151b5bc1d04d5505366c6e27df30af5b

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-methods-from-detached-frame-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-methods-from-detached-frame.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-slice-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-slice-overflow-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream-byob-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream-sync-xhr-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any.js:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/Blob-stream.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/blob/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/File-constructor-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/send-file-formdata-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/file/send-file-formdata-utf-8-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/fileReader-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/Determining-Encoding-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/FileReader-event-handler-attributes-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/FileReader-multiple-reads-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_abort-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_error-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsArrayBuffer-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsBinaryString-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsDataURL-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readAsText-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_readystate-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/reading-data-section/filereader_result-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/sandboxed-iframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url-format.any.js:
(test.t.t.add_cleanup):
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url-with-fetch.any.js:
(async garbageCollect.else.gcRec): Deleted.
(async garbageCollect): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/url_xmlhttprequest_img-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/279121@main
